osu.Game/Screens/Select/Details/AdvancedStats.cs

```diff
@@ -99,6 +102,32 @@ protected override void LoadComplete()
         private ModSettingChangeTracker modSettingChangeTracker;
         private ScheduledDelegate debouncedStatisticsUpdate;
 
+        private StarDifficulty latestStarDifficulty = new StarDifficulty();
+        private void updateBindedInfo()
+        {
+            if (adjustedInfo == null) return;
+
+            // sadly need to calculate this to prevent additional data transportation
+            double rate = 1;
+            foreach (var mod in mods.Value.OfType<IApplicableToRate>())
+                rate = mod.ApplyToRate(0, rate);
+
+            double bpm = 0;
+            if (beatmapInfo != null) bpm = beatmapInfo.BPM * rate;
+
+            BeatmapShortInfo adjusted = new BeatmapShortInfo()
+            {
+                CircleSize = FirstValue.Value.adjustedValue ?? FirstValue.Value.baseValue,
+                DrainRate = HpDrain.Value.adjustedValue ?? HpDrain.Value.baseValue,
+                ApproachRate = ApproachRate.Value.adjustedValue ?? ApproachRate.Value.baseValue,
+                OverallDifficulty = Accuracy.Value.adjustedValue ?? Accuracy.Value.baseValue,
+                BPM = bpm,
+                StarDifficulty = latestStarDifficulty
+            };
+
+            adjustedInfo.Value = adjusted;
+        }
```

Comment:
None of this logic can be here. To see why, perform the following exercise:

- Enter song select
- Switch active tab on the left to any leaderboard (local/global/friends/whatever)
- Open mod overlay
- The display will not update anymore, regardless of what you do

All of this should be implemented locally to the mod overlay. This also likely fixes issues like the tests failing because they don't have `Bindable<BeatmapShortInfo>` registered as a dependency or whatever.

Reply:
Is calculating this locally gonna grab the Star Rating value from cache without calculating it twice?

Reply:
fixed